RELEASE.md

```diff
@@ -1,9 +1,42 @@
-# Release Process
+# Release Guide
 
-The Kubernetes Template Project is released on an as-needed basis. The process is as follows:
+## Semi-automatic
 
-1. An issue is proposing a new release with a changelog since the last release
-1. All [OWNERS](OWNERS) must LGTM this release
-1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
-1. The release issue is closed
-1. An announcement email is sent to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] kubernetes-template-project $VERSION is released`
+1. Make sure your repo is clean by git's standards
+2. Create a release branch `git checkout -b release-1.19` (not required for patch releases)
```

Comment:
This should refer to the OWNERS, as the original doc stated. Very few people have rights to do this, AFAIK.